tools/driver/p4c_src/main.py

```diff
@@ -224,39 +307,152 @@ def main():
+    ### overriding the “should_not_check_input” method.
     checkInput = not (opts.help_pragmas or backend.should_not_check_input(opts))
 
-    input_specified = False
-    if opts.source_file:
-        input_specified = True
-    if (os.environ['P4C_BUILD_TYPE'] == "DEVELOPER"):
-        if opts.json_source:
-            input_specified = True
-    if checkInput and not input_specified:
-        parser.error('No input specified.')
-    elif not checkInput:
-        opts.source_file = "dummy.p4"
-
-    if (os.environ['P4C_BUILD_TYPE'] == "DEVELOPER"):
-        if opts.json_source:
-            opts.source_file = opts.json_source
-
-    if checkInput and not os.path.isfile(opts.source_file):
-        print('Input file {} does not exist'.format(opts.source_file),
-              file = sys.stderr)
-        sys.exit(1)
+    use_a_dummy_P4_pseudoPathname = False
+    ### a note from Abe: do we really still need the dummy pathname?
+    ###                  Maybe we can take all the dummy-related code out once
+    ###                    my work to improve the pathname handling in the
+    ###                    driver is good enough for merging to “main”.
+    if checkInput:
+        if not any_input_specified:
+            parser.error("no input pathname was specified.")
+            ### reminder: at this point, the program is dead
+
+        if len(opts.P4_source_files) > 1:
+            print("\n"
+                  "Sorry; as of this writing, the P4 compiler driver does "
+                  "_not_ support multiple top-level P4 source files in a "
+                  "single invocation.  Multiple P4 source files at a time "
+                  'are currently only supported via "#include"'
+                  "(i.e. additional non-top-level P4 source files).  "
+                  "Number of top-level P4 source-file pathnames detected: "
+                  "" + str(len(opts.P4_source_files)), file=sys.stderr)
+            ###   ^^ this nonsense is to get rid of a W504 in “pycodestyle”
+            ###      [“W504 line break after binary operator”]
+            error_count += 1
+
+        ### We need to do the next 3 lines of code this way (instead of
+        ###   iterating over a concise “list if list else []” expression)
+        ###   because “opts.json_source” might be invalid
+        ###   [i.e. we are _not_ in a developer build].
+        pathnames_to_check = opts.P4_source_files.copy()
+        ### Abe`s reminder to self re the preceding line:
+        ###  a plain “pathnames_to_check = opts.P4_source_files”
+        ###  creates an _alias_!!!
+        if JSON_input_specified:
+            pathnames_to_check.append(opts.json_source)
+
+        for pathname in pathnames_to_check:
+            if not os.path.isfile(pathname):
+                print('\nERROR: the input file "{}" does not exist.'.
+                      format(pathname), file=sys.stderr)
+                error_count += 1
+
+    elif not any_input_specified:
```

Comment:
Maybe you should call this "some_input_specified"

Reply:
Do you feel strongly that the variable must be renamed?